examples/sign-leaf-with-ca.rs

```diff
@@ -0,0 +1,33 @@
+use rcgen::{BasicConstraints, Certificate, CertificateParams, IsCa};
+use std::fs;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+	let mut ca_params: CertificateParams = Default::default();
```

Comment:
I think we should aim to have this example produce an issuer certificate that tries to meet the requirements of RFC 5280 and the CABF baseline requirements. I don't think the Rcgen defaults do a great job of this presently, but we can at least account for this in the example code.

In particular I think we should:

* Include a country name and organization name in the root certificate subject [c.f. baseline reqs "7.1.2.10.2 CA Certificate Naming"](https://github.com/cabforum/servercert/blob/main/docs/BR.md#712102-ca-certificate-naming).

* Include a `keyUsage` extension with `digitalSignature`, `keyCertSig`n and `cRLSign` asserted. [c.f. baseline reqs "7.1.2.10.7 CA Certificate Key Usage"](https://github.com/cabforum/servercert/blob/main/docs/BR.md#712107-ca-certificate-key-usage)

* Fix the validity period so that the `notBefore` is set to one day in the past (instead of a date in 1975), and fix the `notAfter` so that it is no more than 25 years in the future (instead of a date in 4096) c.f. baseline reqs [7.1.2.1.1 Root CA Validity](https://github.com/cabforum/servercert/blob/main/docs/BR.md#71211-root-ca-validity)